modules/nlu/src/views/index.jsx

```diff
@@ -103,51 +110,49 @@ export default class Module extends React.Component {
 
   getFilteredIntents() {
     return this.getIntents().filter(i => {
-      if (!this.state.showHiddenIntents && isHiddenIntent(i)) {
-        return false
-      }
       if (this.state.filterValue.length && !i.name.toLowerCase().includes(this.state.filterValue.toLowerCase())) {
         return false
       }
       return true
     })
   }
 
+  deleteIntent = intent => {
+    const confirmDelete = window.confirm(`Are you sure you want to delete the intent "${intent}" ?`)
+    if (confirmDelete) {
+      return this.props.bp.axios.delete(`/mod/nlu/intents/${intent}`).then(this.fetchIntents)
+    }
+  }
+
   renderCategory() {
     const intents = this.getFilteredIntents()
 
-    const caret = classnames(style.caret, {
-      [style.inverted]: !this.state.showNavIntents
-    })
-
-    const getClassName = el =>
-      classnames({
-        [style.active]: this.getCurrentIntent() === el
-      })
-
     return (
       <div className={style.intentsContainer}>
-        <div>
-          <span>Intents ({intents.length})</span>
-          <span className={caret} onClick={this.toggleProp('showNavIntents')}>
-            <span className="caret" />
-          </span>
-        </div>
-        <Collapse in={this.state.showNavIntents}>
-          <ul className={style.intentsList}>
+        <div className={style.list}>
+          <ListGroup>
             {intents.map((el, i) => (
-              <li key={i} className={getClassName(el)} onClick={() => this.setCurrentIntent(el.name)}>
+              <ListGroupItem
+                key={`nlu_entity_${el.name}`}
+                className={style.entity}
+                onClick={() => this.setCurrentIntent(el.name)}
+              >
                 {el.name}
                 &nbsp;(
                 {_.get(el, 'utterances.length') || 0})
-              </li>
+                <Glyphicon glyph="trash" className={style.deleteEntity} onClick={() => this.deleteIntent(el.name)} />
+              </ListGroupItem>
             ))}
-          </ul>
-        </Collapse>
+          </ListGroup>
+        </div>
       </div>
     )
   }
 
+  onUtterancesChange = () => {
+    this.checkSync()
```

Comment:
Could use directly this.checkSync in the component, maybe give a more meaningful name to checkSync?